src/Analyzers/CSharp/Analyzers/SimplifyPropertyAccessor/CSharpSimplifyPropertyAccessorDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.SimplifyPropertyAccessor;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class CSharpSimplifyPropertyAccessorDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+{
+    public CSharpSimplifyPropertyAccessorDiagnosticAnalyzer()
+        : base(IDEDiagnosticIds.SimplifyPropertyAccessorDiagnosticId,
+               EnforceOnBuildValues.SimplifyPropertyAccessor,
+               CSharpCodeStyleOptions.PreferSimplePropertyAccessors,
+               new LocalizableResourceString(nameof(CSharpAnalyzersResources.Simplify_property_accessor), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+               new LocalizableResourceString(nameof(CSharpAnalyzersResources.Property_accessor_can_be_simplified), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+    {
+    }
+
+    public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+        => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+    protected override void InitializeWorker(AnalysisContext context)
+    {
+        context.RegisterSyntaxNodeAction(AnalyzeAccessorList, SyntaxKind.AccessorList);
+    }
+
+    private void AnalyzeAccessorList(SyntaxNodeAnalysisContext context)
+    {
+        var option = context.GetCSharpAnalyzerOptions().PreferSimplePropertyAccessors;
+        if (!option.Value || ShouldSkipAnalysis(context, option.Notification))
+            return;
+
+        var accessorList = (AccessorListSyntax)context.Node;
+        var accessors = accessorList.Accessors;
+
+        if (!accessorList.Parent.IsKind(SyntaxKind.PropertyDeclaration) ||
+            !IsValidAccessorList(accessorList))
+        {
+            return;
+        }
+
+        foreach (var accessor in accessors)
+        {
+            // get { return field; }
+            // get => field;
+            if (accessor is { RawKind: (int)SyntaxKind.GetAccessorDeclaration, Body.Statements: [ReturnStatementSyntax { Expression.RawKind: (int)SyntaxKind.FieldExpression }] }
```

Comment:
note: we have a Deconstruct extension that should allow you in *some* cases to do things like `x is AccessorDeclarationSyntax(SyntaxKind.SetAccessorDeclaration)`.  Up to you for which cases are nicer like that.  It helps avoid the `.RawKind = (int)` ugliness.

Reply:
It turns out you can even skip the type name and just have `(SyntaxKind.SetAccessorDeclaration)`. I think for these cases it is ok since the type is obvious anyway